wagtail/search/backends/elasticsearch7.py

```diff
@@ -587,12 +587,13 @@ def _compile_plaintext_query(self, query, fields, boost=1.0):
         if query.operator != "or":
             match_query["operator"] = query.operator
 
-        if boost != 1.0:
-            match_query["boost"] = boost
-
         if len(fields) == 1:
-            return {"match": {fields[0].field_name_with_boost: match_query}}
+            if boost != 1.0 or fields[0].boost != 1.0:
+                match_query["boost"] = boost * fields[0].boost
```

Comment:
I'm not really sure where this bit of math comes from...?

Reply:
Boost values can either come from the `SearchField` definition, or [a `Boost` clause within the query](https://docs.wagtail.org/en/stable/topics/search/searching.html#complex-search-queries) - in the latter case, the boost value comes through as the `boost` argument to `_compile_query` and is passed down through recursive calls to `_compile_query`. When boosts exist on both the field and the query, we multiply them to combine their effect - this is consistent with how we handle expressions like `Boost(Boost("query", 2.0), 2.0)` here: https://github.com/wagtail/wagtail/blob/b14bc33cdc93b06257d660256e410dada3beb6ad/wagtail/search/backends/elasticsearch7.py#L677-L678